pyatrea/__init__.py

```diff
@@ -82,328 +85,390 @@ def decompress(self, s):
+                value /= params["coefs"][key]
             return value
         return None
-    
+
     def auth(self):
-        magic = hashlib.md5(("\r\n"+self.password).encode('utf-8')).hexdigest()
-        response = requests.get('http://'+self.ip+'/config/login.cgi?magic='+magic+'&'+random.choice(string.ascii_letters)+random.choice(string.ascii_letters))
-        if(response.status_code == 200):
+        magic = hashlib.md5(("\r\n" + self.password).encode("utf-8")).hexdigest()
+        response = requests.get(
+            "http://"
+            + self.ip
+            + "/config/login.cgi?magic="
+            + magic
+            + "&"
+            + random.choice(string.ascii_letters)
+            + random.choice(string.ascii_letters)
+        )
+        if response.status_code == 200:
             xmldoc = ET.fromstring(response.content)
-            if(response.content == "denied"):
+            if response.content == "denied":
                 return False
             else:
                 self.code = xmldoc.text
         return False
-    
+
     def setPower(self, power):
         try:
             power += 1
         except TypeError:
             return False
         power -= 1
 
-        if(power < 0 or power > 100):
+        if power < 12 or power > 100:
             return False
 
-        self.setCommand('H10708', power)
-        self.setCommand('H01020', power)
+        self.setCommand("H10708", power)
+        self.setCommand("H01020", power)
         return True
 
     def exec(self):
-        url = 'http://'+self.ip+'/config/xml.cgi?auth='+self.code
-        
-        if(len(self.commands) > 0):
+        url = "http://" + self.ip + "/config/xml.cgi?auth=" + self.code
+
+        if len(self.commands) > 0:
             for register in self.commands:
                 url = url + "&" + register + self.commands[register]
             response = requests.get(url)
             return response.status_code == 200
         return False
-    
+
     def setTemperature(self, temperature):
         try:
             temperature += 1
         except TypeError:
             return False
         temperature -= 1
 
-        if(temperature >= 10 and temperature <= 40):
-            self.setCommand('H10710', temperature)
-            self.setCommand('H01021', temperature)
+        if temperature >= 10 and temperature <= 40:
+            self.setCommand("H10710", int(temperature * 10))
+            self.setCommand("H01021", temperature)
             return True
         return False
 
     def setCommand(self, id, value):
         params = self.getParams()
-        if(id in params['ids']):
-            if(id in params['coefs']):
-                value = int(value * params['coefs'][id])
-            if(id in params['offsets']):
-                value = int(value + params['offsets'][id])
-            self.commands[id] = f'{value:05}'
-    
+        status = self.getStatus()
+        if id in params["ids"] or id in status:
```

Comment:
Does your unit support some values which are present in `status` but not in `params`? My understanding is that `params` basically declares all possible variables, while `status` lists their values, so it'd be weird to have a value without declaration.

Reply:
Yes it does, the opposite is true actually for me. See comments on upstream pr, in the case of power for example, there are a few values which change but none of them is in param.

Reply:
Ok, keeping as is. Really weird inconsistency by their API though.